File: programmers/김나영/1/붕대 감기.java
class Solution {
    public int solution(int[] bandage, int health, int[][] attacks) {
        int answer = 0;
        int playerHealth = health;
        int time = 0;
        for (int i = 0; i < attacks.length; i++) {
            int attackTime = attacks[i][0];
            int attackHealth = attacks[i][1];
            
            int leftTime = attackTime - time - 1;
            //공격받기 전까지 체력회복 계산
            if (leftTime > 0) {
                int curHealth = playerHealth + leftTime * bandage[1];
                if (attackTime - time > bandage[0]) {
                    // 연속 성공
                    curHealth += (leftTime/bandage[0])*bandage[2];
                }
                
                if (curHealth < health) playerHealth = curHealth;
                else playerHealth = health;
            }
            
            //공격
            playerHealth -= attackHealth;
            time = attackTime;

            if (playerHealth <= 0) {
                return -1;
            }
        }
        return playerHealth;
    }
}

File: programmers/박소민/lv1/붕대감기.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

/*
설계
시뮬레이션
기본적인 상태와 값들이 주어지고
attacks 배열을 순회하면서 상태를 변화하기 때문.
*/

int solution(vector<int> bandage, int health, vector<vector<int>> attacks) {
    const int t = bandage[0];
    const int x = bandage[1];
    const int y = bandage[2];
    
    int curr_health = health;
    int curr_time = 1;
    int recover_time = 0;
    
    for (const auto& attack : attacks) {
        int attack_time = attack[0];
        int attack_amount = attack[1];
        
        // 1. 공격 시간 전까지 체력 회복
        recover_time = (attack_time - curr_time);
        
        if (recover_time > 0) {
            int total_heal = (recover_time * x) + ((recover_time / t) * y);
            curr_health = min(health, curr_health + total_heal);
        }
        
        // 2. 공격 받은 후 체력 감소
        curr_health -= attack_amount;
        if (curr_health <= 0) {
            return -1;
        }
        
        // 3. 시간 갱신
        curr_time = attack_time + 1;
    }
    
    return curr_health;
}